src/Exporter/ExporterInterface.php

```diff
@@ -0,0 +1,51 @@
+<?php
+namespace RonRademaker\Exporter\Exporter;
+
+use RonRademaker\Exporter\Data\DataInterface;
+use RonRademaker\Exporter\Option\OptionInterface;
+
+/**
+ * API for exporters
+ *
+ * @author Ron Rademaker
+ */
+interface ExporterInterface
+{
+    /**
+     * Create a new exporter with $options
+     *
+     * @param array $options - an array of OptionInterface of key value pair resolvable as an option
+     */
+    public function __construct(array $options = array());
+
+    /**
+     * Export the data
+     *
+     * @param mixed $input - required unless already set using setInput
+     * @param array $options - an array of OptionInterface of key value pair resolvable as an option
+     */
+    public function export(DataInterface $input = null, array $options = array());
+
+    /**
+     * Explicitly set an option
+     *
+     * @param OptionInterface $option
+     * @param boolean $overwrite - Overwrite existing option if it exists
+     */
+    public function setOption(OptionInterface $option, $overwrite = false);
```

Comment:
Multiple options of the same type should be allowed, example:

```
FileNameOption('myfile');
FileNameOption('myotherfile');
```

Should create two files